### PR TITLE
Use gpu namespace

### PR DESCRIPTION
**Pull request by beaufortfrancois**
WebIDL have now a nice `namespace` that fits exactly what we want for the `gpu` entry point of this API.
See https://heycam.github.io/webidl/#idl-namespaces

### CODE REVIEW

File: design/sketch.webidl [hunk 1]
```diff
@@ -682,12 +682,6 @@ dictionary WebGPUAdapterDescriptor {
...
+namespace gpu {
     Promise<WebGPUAdapter> requestAdapter(WebGPUAdapterDescriptor desc);
-};
-
-// Add a "gpu" member to Window that contains the global instance of a "WebGPU"
-interface mixin WebGPUProvider {
-    [Replaceable, SameObject] readonly attribute WebGPU gpu;
-};
-Window includes WebGPUProvider;
+};
```
Comment (Kangz) [inline design/sketch.webidl]: Can we have an extra newline here?

**Comment by kvark** [comment]
This changes semantics: we had a mixin for `Window` to provide WebGPU instance, which had a method to enumerate adapters and such. Now we just have a global method in the namespace. If this was desired, why did we have the mixin in the first place?

Also minor nit: please rebase instead of having a merge commit at the end.

**Comment by Kangz** [comment]
We didn't have a namespace because I didn't know better when I did the mixin.

**Comment by kainino0x** [comment]
I don't think the method is global, it's still inside the gpu namespace
now, right? (so you still have to `gpu.getAdapters`)

File: design/sketch.webidl (hunk 1 above)
Comment (beaufortfrancois) [reply design/sketch.webidl]: Done.

**Comment by kvark** [comment]
@kainino0x it's no longer a method of the `Window`, which I don't yet know if it's going to be a problem.
Anyhow, would be nice to rebase (to avoid merge commits) and squash a few.

**Comment by Kangz** [comment]
@kvark namespaces default to being on the global object which is `window` for our case.

**Pull request merged**